lib/Dialect/FIRRTL/Transforms/LowerMemory.cpp

```diff
@@ -259,6 +259,11 @@ void LowerMemoryPass::lowerMemory(MemOp mem, const FirMemory &summary,
     auto nla = dyn_cast<HierPathOp>(symbolTable->lookup(nlaSym.getAttr()));
     auto namepath = nla.namepath().getValue();
     SmallVector<Attribute> newNamepath(namepath.begin(), namepath.end());
+    if (!nla.isComponent()) {
+      newNamepath[newNamepath.size() - 1] = hw::InnerRefAttr::get(
+          inst->getParentOfType<FModuleOp>().moduleNameAttr(),
+          inst.getInnerNameAttr());
+    }
```

Comment:
I think this should just duplicate the hierarchical path. Also, isn't this changing the hierarchical path from ending in a module to again ending in a component? This will likely break downstream stuff once we fully migrate over.